backend/fms_core/viewsets/sample_pooled.py

```diff
@@ -0,0 +1,117 @@
+class DerivedSampleSerializer(serializers.ModelSerializer):
+    library = LibrarySerializer(required=False)
+    biosample = BiosampleSerializer(required=False)
+    tissue_source = SampleKindSerializer(required=False)
+    sample_kind=SampleKindSerializer(required=False)
+    project = ProjectSerializer(required=False)
+    
+    class Meta:
+        model = DerivedSample
+        exclude = EXCLUDE_BOOKKEEPING
+        depth = 1
+        
+
+class PooledSampleSerializer(serializers.Serializer):
+    ''' Serializes a DerivedBySample object, representing a pooled sample. 
+        The result is a nested data structure containing all of the information about the
+        pooled sample needed by the frontend for display.
+    '''
+    volume_ratio = serializers.DecimalField(max_digits=20, decimal_places=3, read_only=True)
+    derived_sample = DerivedSampleSerializer()
+    # Since DerivedBySample doesn't have its own id field, we use the derived_sample id
+    # as a top level id in the returned data structure. The UX needs this for 'objectsById' stuff.
+    id = serializers.IntegerField(read_only = True, source='derived_sample.id')
+    parent_sample_name=serializers.SerializerMethodField()
+    parent_sample_id = serializers.SerializerMethodField()
+    class Meta:
+        model = DerivedBySample
+        exclude = EXCLUDE_BOOKKEEPING + ['sample']
+
+    # Finds the id of the parent sample from which this pooled sample was derived. For example, if this
+    # pool member is from a library then it returns the id of the library sample. 
+    def get_parent_sample_id(self, obj):
+        parent_sample = Sample.objects.get(parent_of=obj.sample.id, derived_samples=obj.derived_sample.id)
+        return parent_sample.id if parent_sample is not None else ''
+
+    # Finds the id of the parent sample from which this pooled sample was derived.
+    def get_parent_sample_name(self, obj):
+        sample = Sample.objects.get(parent_of=obj.sample.id, derived_samples=obj.derived_sample.id)
+        return sample.name if sample is not None else ''
+
+
+class MissingPoolIDException(APIException):
+    status_code = 400
+    default_code = 'bad_request'
+    default_detail = 'No pool ID: query must include sample__id parameter containing the pool ID.'
+
+class PooledSamplesViewSet(viewsets.ModelViewSet):
+    '''
+        Lists the samples that are contained in a pool. This is a custom endpoint designed
+        for the frontend to display pooled samples in a table. It returns the list of derived
+        samples contained in a single pool.
+
+        The request must include a 'sampled__id__in` query parameter specifying the id of the
+        pool sample. If not, MissingPoolIDException is raised.
+    '''
+    queryset = DerivedBySample.objects.all()
+    serializer_class = PooledSampleSerializer
+    filter_class = PooledSampleFilter
+
+    def get_queryset(self):
+        # Ensure that the pool id is specified to avoid trying to return all of the derived
```

Comment:
I think that to retrieve the list of samples that are pools it would make more sense to add that capability to the Sample viewset (eg. add an 'is_pool' filter option to return samples that are pools). If the user wants to get the set of samples contained by a pool then they would use this endpoint.

Reply:
feel free to add a default filter on "is_pool" in get_queryset since this endpoint is intended to return pool information. I would not set it as an option though as this is really not useful for anything that is not a pool (it will force us to support code that is not useful).